platform/spire/src/virt.py

```diff
@@ -18,10 +18,17 @@
 
 
 def get_bridge(ip):
+    """The name to be used for the bridge interface connecting the VMs and the host."""
     return "spirebr%s" % ip.packed.hex().upper()
 
 
+def get_bridge_access(ip):
+    """The name to be used for the vlan interface providing host-side access to the VMs."""
+    return "spireac%s" % ip.packed.hex().upper()
+
+
 def get_node_tap(node):
+    """The name to be used for the tap interface providing network to a particular QEMU VM."""
```

Comment:
I think I'm a little lost here.  Can you explain what the effect of this network configuration is?

Reply:
In the previous model, we had a bridge on the host and a bunch of tap interfaces associated with the bridge. Each VM would be connected to a tap interface by QEMU. We'd configure an IP address on the bridge so that we could access the VM network, and set up forwarding so that the VMs on the network could access the public internet through the host.

We don't need any change to the bridge or the tap interfaces to support passing VLAN trunk traffic between the VMs; we just need to make sure that the host's bridge IP address is on a VLAN instead of on the bare bridge. To do that, we need to add a subinterface on the bridge (not an interface attached to the bridge, but a VLAN interface on the bridge) that uses the correct VLAN number and bind our IP address to that instead of the bridge directly.

This way, the VMs can't access the network with the host as gateway unless they set up VLAN interfaces.

So the tap interfaces function exactly the same as they did before, and we just add an extra vlan interface to use for binding instead of the bridge when VLAN trunking is enabled.